packages/react/src/index.d.ts

```diff
@@ -4,6 +4,9 @@ declare const UNDEFINED_VOID_ONLY: unique symbol;
 // Destructors are only allowed to return void.
 type Destructor = () => void | { [UNDEFINED_VOID_ONLY]: never };
 
+type InferNone<T> = T extends undefined ? typeof React.None : never;
+type MapToNone<T> = { [K in keyof T]-?: NonNullable<T[K]> | InferNone<T[K]> };
```

Comment:
I don't think the required modifier `-?:` should be here, since [`setState` accepts partial state objects](https://legacy.reactjs.org/docs/state-and-lifecycle.html#state-updates-are-merged). That said, this is a wonderful contribution 👍

Reply:
Done!

Reply:
When we remove -? from the MapToNone type definition, it allows setting optional properties to undefined. This can be demonstrated with the following example:
```ts
import React from ".";

interface State {
	test?: string;
}

export class Test extends React.Component<{}, State> {
	componentDidMount(): void {
		// With -? removed, this is possible:
		this.setState({ test: undefined });
	}
}
```